common/display/displayqueue.cpp

```diff
@@ -713,22 +713,13 @@ void DisplayQueue::SetColorTransform(const float *matrix, HWCColorTransform hint
 
   if (hint == HWCColorTransform::kArbitraryMatrix) {
     memcpy(color_transform_matrix_, matrix, sizeof(color_transform_matrix_));
-    if (*(uint32_t*)(matrix + 12) != 0 ||
-        *(uint32_t*)(matrix + 13) != 0 ||
-        *(uint32_t*)(matrix + 14) != 0) {
       compositor_.SetColorTransformMatrix(color_transform_matrix_);
       state_ |= kDisableOverlayUsage;
```

Comment:
hmm I dont think we need to disable overlay usage in this case. What we can do is force a GPU composition pass for layers in this case. You can do it by adding another state, POST_PROCESSING_FOR_COLOR_MATRIX and here https://github.com/intel/IA-Hardware-Composer/blob/master/common/display/displayqueue.cpp#L454, we can check if that state is set and if so change the state of all composition planes to be gpu rendered i.e. calling this API for all current composition planes https://github.com/intel/IA-Hardware-Composer/blob/master/common/display/displayplanestate.h#L122

Note that when this Matrix changes i.e in case Display can handle it straight away and dont need compositor to get involved, than we will need to force re-validation of layers. This can be done by setting state_ |= FrameStateTracker::kRevalidateLayers;

Reply:
@kalyankondapally Thanks, I'll enhance the implementation according to your comment, and I'll continue followup the feature in PRC holidays.